### PR TITLE
Add balanced fill tracking to avoid unecessary re-randomization

### PR DESCRIPTION
**Pull request by Reidmcc**
This implements a fill tracker for the balanced strategy that simply toggles a boolean for whether to regen levels. If no fills happen the existing levels will be re-submitted.

Addresses https://github.com/interstellar/kelp/issues/52

### CODE REVIEW

File: plugins/balancedLevelProvider.go [hunk 1]
```diff
@@ -27,14 +27,21 @@ type balancedLevelProvider struct {
 	virtualBalanceBase            float64 // virtual balance to use so we can smoothen out the curve
 	virtualBalanceQuote           float64 // virtual balance to use so we can smoothen out the curve
 	orderConstraints              *model.OrderConstraints
+	needNewLevels                 bool // boolean for whether to generate levels, starts true
```
Comment (nikhilsaraf) [inline plugins/balancedLevelProvider.go]: can use a more terse name: `shouldRefresh`

File: plugins/balancedLevelProvider.go [hunk 2]
```diff
@@ -95,35 +105,19 @@ func validateSpread(spread float64) {
...
+		log.Println("no offers were taken, leave levels as they are")
+		return p.lastLevels, nil
+	}
 
-		// always take a spread off the amountCarryover
-		amountCarryoverSpread := p.getRandomSpread(p.minAmountCarryoverSpread, p.maxAmountCarryoverSpread)
-		amountCarryover *= (1 - amountCarryoverSpread)
+	levels, e := p.recomputeLevels(maxAssetBase, maxAssetQuote)
+	if e != nil {
+		log.Fatalf("unable to generate new levels: %s\n", e)
```
Comment (nikhilsaraf) [inline plugins/balancedLevelProvider.go]: we should **never** use log.Fatal
instead we should wrap the error like this: `fmt.Errorf("unable to generate new levels: %s", e)`.
The framework will handle retries etc.

**Comment by Reidmcc** [comment]
@nikhilsaraf updates made

**Comment by nikhilsaraf** [comment]
@Reidmcc made some updates and merging.

**Pull request merged**